pkg/cmd/local/deploy.go

```diff
@@ -0,0 +1,194 @@
+package local
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/go-playground/validator/v10"
+	"github.com/spf13/cobra"
+
+	"github.com/instill-ai/cli/internal/config"
+	"github.com/instill-ai/cli/pkg/cmd/instances"
+	"github.com/instill-ai/cli/pkg/cmdutil"
+	"github.com/instill-ai/cli/pkg/iostreams"
+)
+
+const (
+	DefUsername = "admin"
+	DefPassword = "password"
+)
+
+type DeployOptions struct {
+	IO             *iostreams.IOStreams
+	Exec           ExecDep
+	OS             OSDep
+	Config         config.Config
+	MainExecutable string
+	Interactive    bool
+	Path           string `validate:"required,dirpath" example:"/home/instill-core/"`
+	Branch         string `validate:"required" example:"main"`
+}
+
+func NewDeployCmd(f *cmdutil.Factory, runF func(*DeployOptions) error) *cobra.Command {
+	opts := &DeployOptions{
+		IO: f.IOStreams,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "deploy",
+		Short: "Deploy a local Instill Core instance",
+		Example: heredoc.Doc(`
+			# deploy to /home/me/instill
+			$ inst local deploy --path /home/me/instill
+		`),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cfg, err := f.Config()
+			if err != nil {
+				return err
+			}
+			opts.Config = cfg
+
+			if opts.IO.CanPrompt() {
+				opts.Interactive = true
+			}
+
+			opts.MainExecutable = f.Executable()
+			if runF != nil {
+				return runF(opts)
+			}
+
+			return runDeploy(opts)
+		},
+	}
+	pwd, err := os.Getwd()
+	if err != nil {
+		logger.Error("Couldn't get pwd", err)
+	}
+	dir := filepath.Join(pwd, "instill-core") + string(os.PathSeparator)
+	cmd.Flags().StringVarP(&opts.Path, "path", "p", dir, "Destination directory")
+	cmd.Flags().StringVarP(&opts.Branch, "branch", "b", "main", "Source branch, used to test new features")
+
+	return cmd
+}
+
+func runDeploy(opts *DeployOptions) error {
+	// init, validate
+	var err error
+	path := opts.Path
+	io2 := opts.IO
+	start := time.Now()
+	err = validator.New().Struct(opts)
+	if err != nil {
+		return fmt.Errorf("ERROR: wrong input, %w", err)
+	}
+
+	// check the deps
+	apps := []string{"docker", "make", "git"}
+	for _, n := range apps {
+		if opts.Exec != nil {
+			_, err = opts.Exec.LookPath(n)
+		} else {
+			_, err = opts.Exec.LookPath(n)
```

Comment:
This line causes
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x180646a]
```

Reply:
Please include a full stack trace when reporting issues.